test/api/liskApi.js

```diff
@@ -1099,4 +927,93 @@ describe('Lisk.api()', () => {
 	});
+
+	describe('#createRequestObject', () => {
+		let options;
+		let LSKAPI;
+		beforeEach(() => {
+			options = { limit: 5, offset: 3, details: 'moredetails' };
+			LSKAPI = liskApi({ node: 'localhost' });
+		});
+
+		it('should create a valid request Object for GET request', () => {
+			const requestObject = privateApi.createRequestObject.call(LSKAPI, GET, 'transaction', options);
+			const expectedObject = {
+				method: GET,
+				url: 'http://localhost:8000/api/transaction?limit=5&offset=3&details=moredetails',
+				headers: {
+					'Content-Type': 'application/json',
+					nethash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					broadhash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					os: 'lisk-js-api',
+					version: '1.0.0',
+					minVersion: '>=0.5.0',
+					port: 8000,
+				},
+				body: {},
+			};
+
+			(requestObject).should.be.eql(expectedObject);
+		});
+
+		it('should create a valid request Object for POST request', () => {
+			const requestObject = privateApi.createRequestObject.call(LSKAPI, POST, 'transaction', options);
+			const expectedObject = {
+				method: POST,
+				url: 'http://localhost:8000/api/transaction',
+				headers: {
+					'Content-Type': 'application/json',
+					nethash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					broadhash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					os: 'lisk-js-api',
+					version: '1.0.0',
+					minVersion: '>=0.5.0',
+					port: 8000,
+				},
+				body: { limit: 5, offset: 3, details: 'moredetails' },
+			};
+
+			(requestObject).should.be.eql(expectedObject);
+		});
+
+		it('should create a valid request Object for POST request without options', () => {
+			const requestObject = privateApi.createRequestObject.call(LSKAPI, POST, 'transaction');
+			const expectedObject = {
+				method: POST,
+				url: 'http://localhost:8000/api/transaction',
+				headers: {
+					'Content-Type': 'application/json',
+					nethash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					broadhash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					os: 'lisk-js-api',
+					version: '1.0.0',
+					minVersion: '>=0.5.0',
+					port: 8000,
+				},
+				body: {},
+			};
+
+			(requestObject).should.be.eql(expectedObject);
+		});
+
+		it('should create a valid request Object for undefined request without options', () => {
+			const requestObject = privateApi.createRequestObject.call(LSKAPI, undefined, 'transaction');
+			const expectedObject = {
+				method: undefined,
+				url: 'http://localhost:8000/api/transaction',
+				headers: {
+					'Content-Type': 'application/json',
+					nethash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					broadhash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					os: 'lisk-js-api',
+					version: '1.0.0',
+					minVersion: '>=0.5.0',
+					port: 8000,
+				},
+				body: {},
```

Comment:
Now that we have several tests I would rather have a default `expectedObject` which is extended/overridden as necessary.